osmoutils/accum/accum.go

```diff
@@ -0,0 +1,98 @@
+package accum
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gogo/protobuf/proto"
+)
+
+// We keep this object as a way to interface with the methods, even though
+// only the Accumulator inside is stored in state
+type AccumulatorObject struct {
+	// Store where accumulator is stored
+	Store store.KVStore
+
+	// Accumulator's name (pulled from AccumulatorContent)
+	Name string
+
+	// Accumulator's current value (pulled from AccumulatorContent)
+	Value sdk.Dec
+}
+
+// Makes a new accumulator at store/accum/{accumulator_name}
+// returns an error if already exists / theres some overlapping keys
+func MakeAccumulator(accum_store store.KVStore, accum_name string) error {
+	keybz := []byte(accum_name)
+	if accum_store.Has(keybz) {
+		return errors.New("Accumulator with given name already exists in store")
```

Comment:
nit: Go does not perform redundant reallocations when casting is done directly via argument 
```suggestion
	if accum_store.Has([]byte(accum_name)) {
		return errors.New("Accumulator with given name already exists in store")
```